CMakeLists.txt

```diff
@@ -110,6 +109,19 @@ foreach(backend ${backends})
   list(APPEND nix_INCLUDES ${backend_INCLUDES})
 endforeach()
 
+
+### AUTO-GENERATED VERSION HEADER
+
+configure_file (
+  "${PROJECT_SOURCE_DIR}/version.h.in"
+  "include/nix/libversion.hpp"
+  )
+ 
+# add the binary tree to the search path for include files
+# so that we will find version.h
+include_directories("${PROJECT_BINARY_DIR}/include/nix")
```

Comment:
What if we are change that to `include_directories("${PROJECT_BINARY_DIR}/include/")` and then later on include `<nix/libversion.hpp>`.